source/dynamical_systems/README.md

```diff
@@ -1,2 +1,209 @@
+can be constructed as follows:
+
+```c++
+StateRepresentation::CartesianState csA("A");
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(csA);
+
+StateRepresentation::JointState jsB("B");
+DynamicalSystem::Linear<StateRepresentation::JointState> linear(jsB);
+```
+
+### Configuring the Linear DS
+
+To change the strength of the attractor, a gain can be passed as the second argument during construction or
+passed to the `set_gain()` member function. The gain defines the proportionality between
+a distance unit and a velocity unit, and is internally stored as a square matrix with a size corresponding
+to the degrees of freedom in the state representation. For example, the `CartesianState` has six degrees of freedom
+(XYZ in linear and angular space), while the `JointState` would have as many degrees of freedom as joints.
+The gain can be defined as a matrix directly, as a diagonal vector of the appropriate
+length, or as a scalar (which sets the value along the diagonal elements of the matrix).
+```c++
+// set a gain (scalar, vector or matrix during construction)
+double gain = 10;
+StateRepresentation::CartesianState csA("A");
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(csA, gain);
+
+// or set / update the gain for the created object
+std::vector<double> gains = {1, 2, 3, 4, 5, 6};
+linear.set_gain(gains);
+
+// update the attractor
+StateRepresentation::CartesianState csB("B");
+linear.set_attractor(csB);
+```
+
+### Evaluating the Linear DS
+
+To get the velocity from a state, simply call the `evaluate()` function.
+
+```c++
+StateRepresentation::CartesianState csA("A"), csB("B");
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(csA);
+
+// note: the return type of evaluate is a CartesianState, but
+// it can be directly assigned to a CartesianTwist because the =operator
+// has been defined for that purpose
+StateRepresentation::CartesianState twist = linear.evaluate(csB);
+```
+
+The returned velocity will always be expressed in the same reference frame as the input state.
+
+### Reference frames
+
+The following section applies for the `CartesianState` type.
+
+The `evaulate()` function will always return a twist expressed in the same reference frame as the input state,
+provided that the input state is compatible with the DS.
+
+The input state must be expressed in one of two supported reference frames:
+1. The reference frame of the input is the DS base frame
+2. The reference frame of the input matches the reference frame of the DS base frame
+
+The following snippet illustrates the difference in these two options.
+```c++
+// create a linear DS with attractor B in frmae A
+StateRepresentation::CartesianState BinA("B", "A");
+DynamicalSystems::Linear<StateRepresentation::CartesianState> linearDS(BinA);
+
+linearDS.get_attractor().get_name();             // "B"
+linearDS.get_attractor().get_reference_frame();  // "A"
+linearDS.get_base_frame().get_name();            // "A"
```

Comment:
I may be wrong but above in Base frame section you said *The DynamicalSystem can be constructed with a state to set the base frame* and here you construct the DS with a state but the base frame is not B in A. What's going on here?

Reply:
I tried to clarify that point in the Base frame section a few lines later

> In most cases, the constructor for the base DynamicalSystem should not be used directly,
 and rather the derived DS classes should construct the base accordingly.
 
But I agree, and I didn't resolve Baptiste's earlier comment, that I can be more clear that the Linear constructor sets the attractor and therefore also the base frame